source/gui/blockAction.py

```diff
@@ -0,0 +1,57 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2022 NV Access Limited
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+import config
+from dataclasses import dataclass
+from enum import Enum
+from functools import wraps
+import globalVars
+from typing import Callable
+import ui
+from gui.message import isModalMessageBoxActive
+
+
+@dataclass
+class _Context:
+	blockActionIf: Callable[[], bool]
+	translatedMessage: str
+
+
+class Context(_Context, Enum):
+	SECURE_MODE = (
+		lambda: globalVars.appArgs.secure,
+		# Translators: Reported when an action cannot be performed because NVDA is in a secure screen
+		_("Action unavailable in secure context"),
+	)
+	WINDOWS_STORE_VERSION = (
+		lambda: config.isAppX,
+		# Translators: Reported when an action cannot be performed because NVDA has been installed
+		# from the Windows Store.
+		_("Action unavailable in NVDA Windows Store version"),
+	)
+	MODAL_DIALOG_OPEN = (
+		isModalMessageBoxActive,
+		# Translators: Reported when an action cannot be performed because NVDA is waiting
+		# for a response from a modal dialog
+		_("Action unavailable as an open dialog requires a response"),
+	)
+
+
+def when(*contexts: Context):
+	"""Returns a function wrapper.
+	A function wrapped with `when` will exit early if any supplied context in `contexts` is active.
+	For example, a function wrapped with `@when(Context.SECURE_MODE)` will return early

```

Comment:
This example usage is not very Pythonic, is it?

Not knowing the background, and assuming some `from gui.blockAction` imports, reading:

`@when(Context.SECURE_MODE)`

would imply that the decorated function will run when the secure context is set, not the other way around.

The solutions that come to mind are either to change the wrapper name to `whenNot`, which takes an already verbose intended syntax and makes it even more annoyingly verbose; or to provide an example of the full intended usage instead.

Also, "early" seems a bit inconclusive.

```suggestion
	For example, a function wrapped with `@blockAction.when(Context.SECURE_MODE)` will return without running
```

Reply:
>Not knowing the background, and assuming some `from gui.blockAction` imports, reading:

I think this argument can be made for any symbols.
Keeping contextual information in the namespace of the import eg `from gui import blockAction`, or performing `from gui.blockAction import when as blockWhen` would be pythonic convention in my experience.
I struggled to find a PEP to address this but consider the same argument for the vague `Context` in this module.
I understand that this is uniquely confusing due to the parent namespace `blockAction` implying the negation of `when`.

Otherwise, I agree with your suggestion for the documentation change.